### DISCUSSION TITLE
running with USRP B210

### DISCUSSION THREAD

**Discussion by easybodge** [herlesupreeth/docker_open5gs#402] [General]
I have a few questions regarding the docker_open5gs project.

Do we still need to manually build the database using the kamdbctl create command, or is this automated within the Docker setup?

I’m planning to set up a 5G network using docker_open5gs, as I’ve already successfully configured a 4G LTE setup and achieved 4G connectivity. Regarding the USRP B210, which is USB-powered, what would you recommend setting for the SRS_GNB_IP parameter?

Additionally, I need to change the DL/UL frequencies of the gNB. In a non-Dockerized solution, I recall these settings being configurable in a specific configuration file. Is it possible to achieve this within your project? If so, could you guide me on how to do so?

**Comment by herlesupreeth** [comment]
> Do we still need to manually build the database using the kamdbctl create command, or is this automated within the Docker setup?

No need, its done automatically

> Regarding the USRP B210, which is USB-powered, what would you recommend setting for the SRS_GNB_IP parameter?

No need to change this parameter unless you are running gNB and 5GC on separate machines

> Additionally, I need to change the DL/UL frequencies of the gNB. In a non-Dockerized solution, I recall these settings being configurable in a specific configuration file. Is it possible to achieve this within your project? If so, could you guide me on how to do so?

You can go ahead and alter docker_open5gs/srsran/gnb.yml (dont alter AMF_IP and SRS_GNB_IP) and re-deploy and changes should take into effect

**Reply by easybodge** [reply]
First of all, thank you for providing such an excellent resource for setting up 4G VoLTE with the provided 4g-volte-deploy.yaml and enb.conf configurations. Following the instructions in your README, I was able to set up the system seamlessly. The USRP shows appropriate DL and UL activity along with the expected RAT, PCI, and RNTI values, indicating that the eNodeB is functioning well.

Additionally, two LTE-capable phones with Sysmocom SIMs have successfully connected to Open5GS. I assigned random MSISDNs as phone numbers within the Open5GS WebUI for testing. The registration process completed without any errors, and everything appears to be configured correctly.

However, I encountered an issue when attempting to make calls using the phones' native dialing applications. Despite both phones being registered and showing connectivity, the calls fail to connect. I attempted to troubleshoot further using sngrep and observed the following:

The REGISTER messages from the phones occasionally reach the server on port 5060, but they do not appear to be passed forward.
My understanding is that, in a correctly functioning setup, these messages should flow through the P-CSCF, I-CSCF, and S-CSCF before reaching the intended destination.
Having prior experience with Kamailio as a SIP proxy, I anticipated that, once registered, calls would flow seamlessly through the IMS core. However, it seems there is a bottleneck or missing configuration preventing the calls from progressing.

Given there are no errors in the logs and the system appears to be set up correctly, I suspect I might be missing a small but crucial detail. Would you be able to provide any guidance or insights on what might be causing this issue? Your expertise would be immensely helpful, and I would deeply appreciate your support in resolving this.

Thank you in advance for your time and for creating such a valuable resource

**Reply by herlesupreeth** [reply]
Please attach a pcap taken on the machine running docker_open5gs (without any capture filters) of the following scenario and post it here:

1. Bring down all services
2. Start tcpdump/wireshark and start the capture
3. Bring up all services, start eNB and attach UEs
4. Attempt to call
5. Stop tcpdump/wireshark
6. Attach the pcap here

**Comment by easybodge** [comment]
I have captured two PCAP files to aid in troubleshooting the call issue between phones. Below is an explanation of the setup, procedure, and observations for your review.

1. open5gs_capture.pcap : On Physical Wi-Fi Network Adapter (Host)
Captured on the physical Wi-Fi network adapter (172.19.202.18) of the host machine, which may establish communication with User Agents (UAs), I used the following command:

**sudo tcpdump -i wlo1 -w - | tee open5gs_capture.pcap | tcpdump -r -**

This allowed me to log packets into the file (open5gs_capture.pcap) while simultaneously observing live traffic in the terminal.

**Observations:**
From this capture, it appears all traffic remains confined within the default subnet where the wlo1 interface operates.

Steps Followed:
Started all EPC services:
docker-compose -f 4g-volte-deploy.yaml up

Started the eNodeB:
docker-compose -f srsenb.yaml up -d && docker container attach srsenb

Phone with MSISDN: 008 automatically connected when the eNodeB became active.
Toggled flight mode on the second phone (MSISDN: 007), and it connected immediately.
Attempted to place a call from MSISDN: 007 to MSISDN: 008, but the call was unsuccessful.
Stopped the capture using:
sudo pkill tcpdump


2. docker_open5gs_default.pcap: Docker Network Bridge (docker_open5gs_default)
Captured on the Docker network bridge interface (br-e2c7ab47e758), I used the command:

**sudo tcpdump -i br-e2c7ab47e758 -w - | tee docker_open5gs_default.pcap | tcpdump -r -**

The sequence of operations was the same as above. I terminated the capture using Ctrl+C, so there may have been packet transfers during the final moments of the capture.

Does your solution on GitHub automatically handle IP routing between the Dockerized execution environment and external devices, such as User Agents and mobile phones? I believe that UE_IPV4_IMS/INTERNET IPs can remain unchanged, provided the local network operates on a different subnet. Will your solution dynamically assign IPs from these subnets to all UAs over private networks?

Both PCAP files (open5gs_capture.pcap and docker_open5gs_default.pcap) are attached for your review. Please let me know if further details are needed.
[open5gs_phone_call.zip](https://github.com/user-attachments/files/18286787/open5gs_phone_call.zip)

**Reply by herlesupreeth** [reply]
Thanks for pcap. I observed that uou are using softphones and not really UEs - this should matter but on Linphone you have hardcoded P-CSCF IP. Rather it should be set as `sip:ims.mnc070.mcc999.3gppnetwork.org`

> Does your solution on GitHub automatically handle IP routing between the Dockerized execution environment and external devices, such as User Agents and mobile phones?

IP address allocation is handled by the Core Network (open5gs PGW)

> I believe that UE_IPV4_IMS/INTERNET IPs can remain unchanged, provided the local network operates on a different subnet.

Yes it can remain unchanged

> Will your solution dynamically assign IPs from these subnets to all UAs over private networks?

IP address is automatically assigned from UE_IPV4_IMS/INTERNET IPs subnet by the open5gs to all UEs which attaches to 4G network

**Reply by easybodge** [reply]
I wanted to reach out regarding the challenges I’m facing in establishing a phone-to-phone call using your Docker project. Specifically, I am using the native dialer app within the phones (not softphones), and despite multiple attempts, I have been unable to connect the calls successfully.

Key Questions:
Phone-to-Phone Video Call Feasibility: Have you used your Docker project to successfully achieve phone-to-phone video calls? If so, could you provide any insights into the configurations or steps required?

Traffic Routing Through IMS and Kamailio:

Typically, on a server running Kamailio, traffic should be reachable by external UAs (User Agents). However, our team has previously struggled to dockerize Kamailio and RTPEngine effectively.
Could you provide guidance on how traffic from a UA would pass through IMS, Kamailio, and RTP, and reach external UAs for a video call scenario?
Interaction/Sequence Diagrams:

Have you created any interaction or sequence diagrams that show IP and service messaging during a call? These would be immensely helpful for us to better understand the flow.

In this setup, communication occurs via the USRP B210 RAN, not through Wi-Fi. Therefore, the wlo1 interface may not be entirely suitable for debugging. Instead, the traffic captured through the Docker bridge interface (docker_open5gs_default) appears more relevant.

I would greatly appreciate your support and guidance on these points, as well as any advice on resolving the issues we’re facing.

Thank you for your time and effort
